packages/auth/src/web-auth.ts

```diff
@@ -0,0 +1,54 @@
+import pify from 'pify';
+import { WebAuth } from 'auth0-js';
+import camelCase from 'camelcase';
+
+const getOptionsFromLocation = (location: Location | URL) =>
+  Object.fromEntries(
+    [...new URLSearchParams(location.search).entries()].map(([key, value]) => [
+      camelCase(key),
+      value,
+    ]),
+  );
+
+type SsoConnection = 'google-oauth2' | 'ORCID';
+
+export const authorizeWithSsoFactory =
+  (webAuth: WebAuth) =>
```

Comment:
so technically this package does not actually need to expose auth0-js dependency, you could instead ask for the config in this factory instead and then instantiate WebAuth with that. Like this

```typescript
export const authorizeWithSsoFactory =
  ({ AUTH0_CLIENT_ID, AUTH0_DOMAIN }: Config) => { 
   const webAuth = new WebAuth({
      clientID: AUTH0_CLIENT_ID,
      domain: AUTH0_DOMAIN,
   }); 
   return (location: Location | URL, connection: SsoConnection): void =>
```